### DISCUSSION TITLE
pretty_print in split

### DISCUSSION THREAD

**Discussion by ranjithshegde** [folke/noice.nvim#141] [Q&A] [answered]
I am not able to understand what properties `lua` messages have to identify in `filter` options.

I want to send `vim.pretty_print` or `:lua = ...` output longer than 20 lines to go into split.

I have tried the presets for split.

I have tried `routes = {filters = {kind = ` with values `echo`, `echomsg`, but neither pick up lua. None of the options in `ui-messages` kind seemed to apply for lua prints

**Comment by folke** [comment]
`config.preset.long_message_to_split=true` :)

**Reply by ranjithshegde** [reply]
This did not work for lua messages.

**Comment by folke** [comment, marked as answer]
Lua prints will haven an `""` empty kind. As do most messages.

**Reply by ranjithshegde** [reply]
This -> setting `kind` to `""` did.

Thanks

**Reply by ranjithshegde** [reply]
I see the problem! 
both `lua = ` and `lua vim.pretty_print` mark each line of the output as a new message. 
So, `min_height` does not work

**Reply by ranjithshegde** [reply]
And if I route all `kind = ""` to split, I will have to filter out so many other messages

**Reply by folke** [reply]
> I see the problem! both `lua = ` and `lua vim.pretty_print` mark each line of the output as a new message. So, `min_height` does not work

Ah right, yeah, that sucks. I have this in my dots that I use all over the place for debugging:

```lua
_G.d = function(...)
  local msg = vim.inspect(...)
  vim.notify(msg, vim.log.levels.INFO, {
    title = "Debug",
    on_open = function(win)
      vim.wo[win].conceallevel = 3
      vim.wo[win].concealcursor = ""
      vim.wo[win].spell = false
      local buf = vim.api.nvim_win_get_buf(win)
      vim.treesitter.start(buf, "lua")
    end,
  })
end
```

**Reply by folke** [reply]
![image](https://user-images.githubusercontent.com/292349/198376253-f9c73f35-4933-4d46-85fb-ffec9fa51cd6.png)

**Reply by folke** [reply]
In Noice, you can then route that message to split with min_height, or with that preset that does it automatically

**Reply by ranjithshegde** [reply]
Yes this is very clever! sending to `notify` and rerouting through noice!!

Thanks again

**Reply by folke** [reply]
If you route it to a split, make sure to set `lang="lua"`, so you get the coloring.
Then of course you dont need that on-open method at all

**Reply by ranjithshegde** [reply]
Route has a lang!! of course it does!!
I love this plugin and your work in general!

**Reply by folke** [reply]
haha, thanks :)
I added the lang support for the cmdline initially, but use it internally for some Noice commands like stats etc

**Reply by ranjithshegde** [reply]
Also, off-topic. Really appreciate your method for `luasnip` signature help.

In my config I was doing all kinds of shenanigans with `ModeChanged` detection (from select-mode to insert-mode to detect jumps),  and separate per-client per-buffer autocmd-ids to prevent double calls and/or accidental focusing.

How you managed it is so much more elegant

**Reply by folke** [reply]
I really like it as well. I was using cmp-signature before, but that didnt have that. So much better now

**Reply by ranjithshegde** [reply]
FYI, the preset does not route long `mini`/`notify` messages as the event is explicitly set to `msg_show`

**Reply by folke** [reply]
right, but probably better that way, since most users will like that preset for the `msg_show` messages.
You can add a similar route. See `config.presets`
